streamingccl: make completeStreamIngestion use jobs.Update API

gh-casper opened the pull request:
Currently, completeStreamIngestion uses direct queries against
jobs table to select the job, and update the job record.
Now we are using jobs.Update api.

Release note: None

Closes: https://github.com/cockroachdb/cockroach/issues/78415

gh-casper committed:
streamingccl: make completeStreamIngestion use jobs.Update API

Currently, completeStreamIngestion uses direct queries against
jobs table to select the job, and update the job record.
Now we are using jobs.Update api.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/80090)
<!-- Reviewable:end -->

miretskiy reviewed (approved):
> Reviewed 1 of 1 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/80090)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @HonoreDB)



<!-- Sent from Reviewable.io -->

gh-casper commented:
bors r+

gh-casper committed:
streamingccl: highwater doesn't have to pass cutover time

Currently we have to let call site to wait until highwater
mark passed the target cutover time before setting it.

Release note: None

gh-casper commented:
bors r+

gh-casper commented:
bors r+
